src/Scope.ml

```diff
@@ -31,6 +32,10 @@ struct
       let init = {visible = init_visible; export = Trie.empty} in
       M.run @@ fun () -> R.run ~env @@ fun () -> S.run ~init f
 
+    let try_with ?(export_prefix = Emp) ?get ?set f =
+      let env = {export_prefix} in
+      M.run @@ fun () -> R.run ~env @@ fun () -> S.try_with ?get ?set f
```

Comment:
```suggestion
    let try_with = S.try_with
```
Unless I miss something... it seems we should not have `M.run` or `R.run` in `try_with`?

Reply:
Hmm.  For my application I want to use `try_with` *without* any enclosing `run`, and `M.run` and `S.run` need to be called *somewhere*.  But I see your point that if `try_with` is being called *inside* a `run` then you don't want to re-run `M` and `S`.

I suppose I could wrap my code in an extra outer `run` just to get `M` and `R` initialized and just ignore the state created by that call to `run`.  It seems a bit kludgy but it would probably work, at least partially.  In principle, though, it seems on further thought like the mutex should actually be attached to my enclosing undo-buffer state module.

Maybe a better solution would be to also write `Mutex.try_with` and `Reader.try_with`, and then expose those here at the same time?

Reply:
I will have to think about it. I start to worry that we are exposing too many internal details and maybe the correct solution is to (1) implement the proper Undo effects and (2) reimplement Scope on top of it.

Reply:
I think a cleaner solution is to revert our recent PR on algaeff and parametrize Scope with another "State" module that implements your get/set. Do you think that would work? If so I would revert the changes on algaeff.

Reply:
Well, I also wanted the algaeff version for some other uses of that module, not just in Scope.  I see the argument that you don't want to expose too many internal details of Scope, but I don't see any problem with `State.try_with`; it has many other potential uses, and as I said I think the ability to override handlers is part of the point of algebraic effects.

Can you show me an example of what a state-parametrized Scope would look like?

Reply:
Maybe I should just give up and copy-and-paste into my own Scope module.  So much of the design of your Scope seems to be mandated by the fact that `section` takes a callback, and I want to do that differently, so maybe I shouldn't really be trying to reuse so much of it.

Reply:
I just realized I do also want an `R.try_with` here.